src/Microsoft.Graph.Core/Requests/Middleware/RetryHandler.cs

```diff
@@ -115,7 +115,7 @@ private async Task<HttpResponseMessage> SendRetryAsync(HttpResponseMessage respo
                 // Call base.SendAsync to send the request
                 response = await base.SendAsync(request, cancellationToken);
 
-                if (!ShouldRetry(response) || !request.IsBuffered() || !RetryOption.ShouldRetry(RetryOption.Delay, retryCount, response))
+                if (retryCount >= RetryOption.MaxRetry || !(request.IsBuffered() && (ShouldRetry(response) || RetryOption.ShouldRetry(RetryOption.Delay, 0, response))))
```

Comment:
I need to correct this, it should be passing `retryCount` as a parameter.

note: apart from `retryCount` check relating to iteration, this check is modified to be the inverse of `SendAsync` retry check logic.

Reply:
the `retryCount` check related part actually need to go as well.. it was too impactful, handling the expected `ServiceException` therefore had to be added.